### ISSUE TITLE
Shouldn't be allowed to search groups if nothing selected

### ISSUE THREAD

**Issue by richarddubay** [NewSpring/Apollos#354]
## Acceptance Criteria

- [ ] It shouldn't allow you to search for groups if there is no specific filter, campus, or zip code selected. One of the three has to be chosen in order for the search to function.

## Context

In the current app if you do not pick a filter (or type one in), a campus, or a zip code then you are not allowed to search for groups. The button to search is disabled. The idea is that it could potentially be too many groups to pull back (since you are literally pulling _every group_). That's a large query that currently breaks the app. 

Part of the reasoning behind having the campus list have an empty slot and not the "All Locations" option was to have a visual way to show that you hadn't selected a campus at all. 

Side Note: It looks like (at least from https://beta-my.newspring.cc/groups) that the campus and zip code don't default to my personal campus and zip code when I am logged in as they do in our current app. That's something to check on anyway. I can do more testing.

**Comment by conrad-vanl** [comment]
I WANT TO SEE EVERY GROUP MAN! 😎

**Comment by cdanwards** [comment]
@conrad-vanl so after investigating this and talking to @richarddubay about it, the button isn't disabled because of the auto-selection of "Search My Location" to yes but in Holtzman this option is defaulted to no. I guess this is a question of should we have this auto-selected to yes or if this should be considered when disabling the button. Any thoughts?

**Comment by conrad-vanl** [comment]
My thoughts on this is that if it's auto selected to no you have the ability to type in a zipcode instead, and auto-selected to yes probably creates a faster experience for most users...

**Comment by cdanwards** [comment]
@IsaacHardy do you have an opinion one way or the other about this?

**Comment by IsaacHardy** [comment]
I'm like the experience @conrad-vanl explained above.

**Comment by cdanwards** [comment]
@conrad-vanl I'm having some issues trying to create the conditional validation with Yup that I think would work with the `disabled={!isValid}` on the submission button.

I'm using 
`
Yup._____().when('useDeviceLocation', { 
   is: false, 
   then: Yup.____().required(), 
   else: Yup.____(),
}
`
But isn't quite what we need for this.

 Is this the right line of thinking or would you go about it another way?

**Comment by cdanwards** [comment]
@conrad-vanl I've been able to implement the linked validation successfully, however I'm struggling with figuring out how to make is so that `(!values.useDeviceLocation).isValid(false)` That's not really the right syntax but that the logic that we need but I can't figure out where it belongs.

**Comment by cdanwards** [comment]
That is correct. I just pushed my branch 354-group-search.

**Comment by conrad-vanl** [comment]
@cdanwards actually - what you had should've been working out of the box. I updated the `formik` dependency which closed up a bug with `.test` not affecting `isValid` properly. I also had to add `nullable` to the `campusId` field validation, as the `campusId` is initially equal to `null`, which isn't a string.

Lastly, I moved the validation schema into its own variable so that I could call it inside of `isInitialValid`. I don't think this was actually necessary - but this is what we're doing on other forms which allows the validation schema to run when the component first mounts, instead of only after you make some changes to it.
